avocado/utils/cpu.py

```diff
@@ -220,12 +220,39 @@ def get_cpuidle_state():
     return cpu_idlestate
 
 
-def set_cpuidle_state(state_number="all", disable=1, setstate=None):
+def _bool_to_binary(value):
+    '''
+    Turns a boolean value (True or False) into data suitable for writing to
+    /proc/* and /sys/* files.
+    '''
+    if value is True:
+        return b'1'
+    if value is False:
+        return b'0'
+    raise TypeError('Value is not a boolean: %s', value)
+
+
+def _legacy_disable(value):
+    '''
+    Support for the original acceptable disable parameter values
+
+    TODO: this should be removed in the near future
+    Reference: https://trello.com/c/aJzNUeA5/
+    '''
+    if value is 0:
+        return b'0'
+    if value is 1:
+        return b'1'
+    return _bool_to_binary(value)
```

Comment:
If this to-be-removed in a near future we need to add `logging.warning` and I don't think adding 2 such simplistic functions for 2/1 usages makes sense. I think 1 fairly simple function would be enough...

Reply:
Actually one can simply use `return b'1' if bool(value) else b'0'`

Reply:
Anyway looking at the module I think the `1` and `0` makes more sense as the whole module uses the actual values to be assigned. So turning them into `bytes`, yes, but changing it to `True/False` I don't think so. @abdhaleegit, @narasimhan-v any remarks here?

Reply:
@ldoktor can you give one example for:

"Anyway looking at the module I think the 1 and 0 makes more sense as the whole module uses the actual values to be assigned".

I found no additional public interface on this module that *takes* those values (1 and 0).  Now, when it comes to returning the values, it's quite a mess.  Look at at `_get_cpu_status()`:

```python
def _get_cpu_status(cpu):
    """
    Check if a CPU is online or offline

    :pram cpu: CPU number 1 or 2 or 39
    :type cpu: integer
    :returns: `bool` True if online or False if not
    :rtype: 'bool'
    """
    with open('/sys/devices/system/cpu/cpu%s/online' % cpu) as online:
        if '1' in online.read():
            return True
    return False
```

Should it return 0/1? I don't think so.  And then `online()` and `offline()` is IMO inconsistent:

```python
def online(cpu):
    """
    Online given CPU
    """
    with open("/sys/devices/system/cpu/cpu%s/online" % cpu, "w") as fd:
        fd.write('1')
    if _get_cpu_status(cpu):
        return 0
    return 1


def offline(cpu):
    """
    Offline given CPU
    """
    with open("/sys/devices/system/cpu/cpu%s/online" % cpu, "w") as fd:
        fd.write('0')
    if _get_cpu_status(cpu):
        return 1
    return 0
```

A sensible interface to the target users is most of the point of an API.  And if you look at how this is API is *used*, it's clear that it's using the wrong type for Python code:

https://github.com/avocado-framework-tests/avocado-misc-tests/blob/master/cpu/em_cpuhotplug.py#L111

Reply:
Then simply remove it, because without any warning users will only learn this when it stops working.

Reply:
Btw this function was not in 52LTS, therefor the API changes are not that problematic...

Reply:
@ldoktor there's no breakage here.  There's explicit support for keeping things working.  Then, those utility API changes are handled by the release notes... Take a look here:

http://avocado-framework.readthedocs.io/en/61.0/ReferenceGuide.html#utility-libraries